src/Fixer/StringNotation/EscapeImplicitBackslashesFixer.php

```diff
@@ -89,67 +97,32 @@ public function getPriority(): int
     }
 
-    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    protected function createProxyFixers(): array
     {
-        static $singleQuotedRegex = '/(?<!\\\\)\\\\((?:\\\\\\\\)*)(?![\\\'\\\\])/';
-        static $doubleQuotedRegex = '/(?<!\\\\)\\\\((?:\\\\\\\\)*)(?![efnrtv$"\\\\0-7]|x[0-9A-Fa-f]|u{)/';
-        static $heredocSyntaxRegex = '/(?<!\\\\)\\\\((?:\\\\\\\\)*)(?![efnrtv$\\\\0-7]|x[0-9A-Fa-f]|u{)/';
-
-        $doubleQuoteOpened = false;
-        foreach ($tokens as $index => $token) {
-            $content = $token->getContent();
-            if ($token->equalsAny(['"', 'b"', 'B"'])) {
-                $doubleQuoteOpened = !$doubleQuoteOpened;
-            }
-            if (!$token->isGivenKind([T_ENCAPSED_AND_WHITESPACE, T_CONSTANT_ENCAPSED_STRING]) || !str_contains($content, '\\')) {
-                continue;
-            }
-
-            // Nowdoc syntax
-            if ($token->isGivenKind(T_ENCAPSED_AND_WHITESPACE) && '\'' === substr(rtrim($tokens[$index - 1]->getContent()), -1)) {
-                continue;
-            }
-
-            $firstTwoCharacters = strtolower(substr($content, 0, 2));
-            $isSingleQuotedString = $token->isGivenKind(T_CONSTANT_ENCAPSED_STRING) && ('\'' === $content[0] || 'b\'' === $firstTwoCharacters);
-            $isDoubleQuotedString =
-                ($token->isGivenKind(T_CONSTANT_ENCAPSED_STRING) && ('"' === $content[0] || 'b"' === $firstTwoCharacters))
-                || ($token->isGivenKind(T_ENCAPSED_AND_WHITESPACE) && $doubleQuoteOpened);
-            $isHeredocSyntax = !$isSingleQuotedString && !$isDoubleQuotedString;
-            if (
-                (false === $this->configuration['single_quoted'] && $isSingleQuotedString)
-                || (false === $this->configuration['double_quoted'] && $isDoubleQuotedString)
-                || (false === $this->configuration['heredoc_syntax'] && $isHeredocSyntax)
-            ) {
-                continue;
-            }
-
-            $regex = $heredocSyntaxRegex;
-            if ($isSingleQuotedString) {
-                $regex = $singleQuotedRegex;
-            } elseif ($isDoubleQuotedString) {
-                $regex = $doubleQuotedRegex;
-            }
-
-            $newContent = Preg::replace($regex, '\\\\\\\\$1', $content);
-            if ($newContent !== $content) {
-                $tokens[$index] = new Token([$token->getId(), $newContent]);
-            }
-        }
+        $stringImplicitBackslashes = new StringImplicitBackslashesFixer();
+        $stringImplicitBackslashes->configure([
+            'single_quoted' => /* $this->configuration['single_quoted'] ? 'escape' : */ 'ignore',
+            'double_quoted' => /* $this->configuration['double_quoted'] ? */ 'escape' /* : 'ignore' */,
+            'heredoc_syntax' => /* $this->configuration['heredoc_syntax'] ? */ 'escape' /* : 'ignore' */,
+        ]);
+
+        return [
+            $stringImplicitBackslashes,
+        ];
```

Comment:
@Wirone how to proxy cnf, ie. how to setup the new fixer using the old/current configuration? The `$this->configuration` is null.

Reply:
In the old fixer (the one that becomes a proxy) you just configure a new fixer with values that represent old behaviour. Please take a look a other fixers that extend `AbstractProxyFixer` and at the same time implement `ConfigurableFixerInterface`, for example [this](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/860da6f34e2f9965aa922d5e0572b3aab4c0baa9/src/Fixer/Phpdoc/PhpdocNoAliasTagFixer.php) 🙂.

PS. You also need to implement `DeprecatedFixerInterface`

Reply:
Thank you a lot for your support!